TUTORIAL.md

```diff
@@ -434,6 +436,20 @@ end
 
 For an argument, each function will be tried in order. The first function that doesn't raise a `ContractError` will be used. So in this case, if x == 1, the first function will be used. For all other values, the second function will be used.
 
+This allows you write methods more declaratively, rather than using conditionals. This feature is not only useful for recursion; you can use it to keep parallel use cases separate:
+
+```ruby
+Contract lambda{|n| n < 12 } => Ticket
+def get_ticket(age)
+  ChildTicket.new(age: age)
+end
+
+Contract lambda{|n| n >= 12 } => Ticket
```

Comment:
Yes, that is too.

I was talking about different thing, for example, second contract could be just `Num` without lambda, because if `n < 12` fails, then it is obviously `n >= 12`, given n is a number. It is like underscore in pattern matching - last pattern catches all cases.

By the way, just to clarify, when pattern matching like this, Contracts are checked in their definition order, and method that will be executed determined by the first successful contract.

Something along these lines:

```ruby
Contract And[Num, lambda { |n| n < 12 } => Ticket
# def ...

Contract Num => Ticket
# def ...
```

On the other hand, I kinda like still having lambda in the second case, because it makes you understand when this pattern will be used without reading throughout all other patterns.

What do you think?